bigtable/tests/unit/test_batcher.py

```diff
@@ -0,0 +1,182 @@
+
+
+import unittest
+
+import mock
+
+from ._testing import _make_credentials
+
+
+class TestMutationsBatcher(unittest.TestCase):
+    from grpc import StatusCode
+
+    TABLE_ID = 'table-id'
+    TABLE_NAME = '/tables/' + TABLE_ID
+
+    # RPC Status Codes
+    SUCCESS = StatusCode.OK.value[0]
+
+    @staticmethod
+    def _get_target_class():
+        from google.cloud.bigtable.table import Table
+
+        return Table
+
+    def _make_table(self, *args, **kwargs):
+        return self._get_target_class()(*args, **kwargs)
+
+    @staticmethod
+    def _get_target_client_class():
+        from google.cloud.bigtable.client import Client
+
+        return Client
+
+    def _make_client(self, *args, **kwargs):
+        return self._get_target_client_class()(*args, **kwargs)
+
+    @staticmethod
+    def _get_target_batcher_class():
+        from google.cloud.bigtable.batcher import MutationsBatcher
+
+        return MutationsBatcher
+
+    def _make_batcher(self, *args, **kwargs):
+        return self._get_target_batcher_class()(*args, **kwargs)
+
+    def _make_rows(self, table):
+        from google.cloud.bigtable.row import DirectRow
+
+        rows = [DirectRow(row_key=b'row_key', table=table),
+                DirectRow(row_key=b'row_key_2', table=table),
+                DirectRow(row_key=b'row_key_3', table=table),
+                DirectRow(row_key=b'row_key_4', table=table)]
+        rows[0].set_cell('cf1', b'c1', 1)
+        rows[1].set_cell('cf1', b'c1', 2)
+        rows[2].set_cell('cf1', b'c1', 3)
+        rows[3].set_cell('cf1', b'c1', 4)
+
+        return rows
+
+    def test_constructor(self):
+        from google.cloud.bigtable.batcher import MutationsBatcher
+
+        credentials = _make_credentials()
+        client = self._make_client(project='project-id',
+                                   credentials=credentials, admin=True)
+
+        instance = client.instance(instance_id='instance-id')
+        table = self._make_table(self.TABLE_ID, instance)
+
+        mutation_batcher = MutationsBatcher(table)
+        self.assertEqual(table, mutation_batcher.table)
+
+    def test_add_row(self):
+        table = _Table(self.TABLE_NAME)
+        mutation_batcher = self._make_batcher(table)
+
+        rows = self._make_rows(table)
+
+        for row in rows:
+            mutation_batcher.mutate(row)
+
+        mutation_batcher.flush()
+
+        self.assertEqual(table.mutation_calls, 1)
+
+    def test_flush_with_no_rows(self):
+        table = _Table(self.TABLE_NAME)
+        mutation_batcher = self._make_batcher(table)
+        mutation_batcher.flush()
+
+        self.assertEqual(table.mutation_calls, 0)
+
+    def test_add_row_with_max_flush_count(self):
+        table = _Table(self.TABLE_NAME)
+        mutation_batcher = self._make_batcher(table, flush_count=3)
+
+        rows = self._make_rows(table)
+
+        for row in rows:
+            mutation_batcher.mutate(row)
+
+        mutation_batcher.flush()
```

Comment:
please remove the extra `mutation_batcher.flush()`.  There's no need for it in this test.

Reply:
@sduskis Done